geochem_inverse_optimize.py

```diff
@@ -245,6 +251,29 @@ def get_unique_upstream_areas(sample_network: nx.DiGraph):
     return areas
 
 
+def calibrate_regularizer(
+    sample_network: nx.DiGraph,
+    element_data: ElementData,
+    min_: float,
+    max_: float,
+    number: float,
+):
+    vals = np.logspace(min_, max_, num=number)  # regularizer strengths to try
+    for val in vals:
+        print("________________________________________________________")
+        print("Trying regularizer strength: 10^", round(np.log10(val), 3))
+        _, _ = sample_network.solve(element_data, solver="ecos", regularization_strength=val)
+        roughness = cp.norm(cp.vstack(sample_network._regularizer_terms)).value
```

Comment:
Using underscored properties outside of their owner is an antipattern since that's an indicator that the thing is private. Consider adding a method to the class that gives you `cp.norm(cp.vstack(sample_network._primary_terms)).value` or what not and using that method here.

This separates the internal implementation of the class from its interface.

Reply:
Thanks for introducing to the concept of an antipattern, and this specific instance. 

I've added `get_roughness` and `get_misfit` methods to the class.